pt_os_web_portal/backend/helpers/vnc_advanced_wifi_gui.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from pitop.common.command_runner import run_command
+from pitop.common.formatting import is_url
+from pt_web_vnc import clients, connection_details, start, stop
+
+from pt_os_web_portal.backend.helpers.vnc import PtWebVncDisplayId
+
+logger = logging.getLogger(__name__)
+
+
+def stop_advanced_wifi_gui():
+    connected_clients = clients(PtWebVncDisplayId.AdvancedWifiGui.value)
+    should_stop_service = connected_clients == 0
+
+    logger.info(
+        f"{'' if should_stop_service else 'Not'} stopping advanced wifi GUI, it has {clients} connected clients."
+    )
+    if should_stop_service:
+        stop(display_id=PtWebVncDisplayId.AdvancedWifiGui.value)
+
+
+def start_advanced_wifi_gui():
+    if is_url(_advanced_wifi_gui_vnc_details().url):
+        logging.info("Advanced wifi GUI is already running, skipping start...")
+        return
+
+    # Determine which advanced wifi connection app to run
+    cmd = "systemctl is-active dhcpcd"
+    if run_command(cmd, check=False, timeout=5).strip() == "active":
+        app = "wpa_gui"
+    else:
+        app = "nm-connection-editor"
+
+    # Start pt-web-vnc
+    logging.info(f"Starting advanced wifi GUI with '{app}'")
+    start(
```

Comment:
slight issue in that this will block, which may block all server requests (unless this uses a concurrent server of some sort eg wsgi)